gapis/perfetto/trace.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perfetto
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync/atomic"
+
+	"perfetto_pb"
+
+	"github.com/google/gapid/core/event/task"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/os/android"
+	"github.com/google/gapid/core/os/android/adb"
+	"github.com/google/gapid/core/os/file"
+	"github.com/google/gapid/gapis/service"
+)
+
+const (
+	// perfettoTraceFile is the location on the device where we'll ask Perfetto
+	// to store the trace data while tracing.
+	perfettoTraceFile = "/data/misc/perfetto-traces/gapis-trace"
```

Comment:
This should probably be attached to the device. Especially if we want to be able to take perfetto traces elsewhere.

Reply:
I've moved this into an `android` subdirectory. PTAL